### DISCUSSION TITLE
Link forcing hard reload

### DISCUSSION THREAD

**Discussion by rodolphonetto** [vercel/next.js#15614] [Help]
Hello guys, I'm facing a problem that all my Link are doing a hard reload on the pages, let me show you. That is just a example but all them are the same and all they do hard reloads =/

```
              <Link
                href={`/[slug]?id=league-of-legends`}
                as={`/league-of-legends`}
                passHref
              >
                <S.ListItem>lol</S.ListItem>
              </Link>
```

S.ListItem is a `<a>` styled componets ok? That ?id is a param I use client side.

I use a server.js with express bacause I need to handle some URL tricks of the old site. I can't change URL estructures.

```
  server.get("/:slug", async (req, res) => {
    const queryParams = {
      slug: req.params.slug,
      from: req.query.from,
      url: req.query.url,
    }
    let pagePath
    if (categories.includes(queryParams.slug)) {
      pagePath = "/category"
    } else if (queryParams.slug == "more-games") {
      pagePath = "/more-games"
    } else {
      pagePath = "/article"
    }
    return ssrCache({ req, res, pagePath, queryParams })
  })
```

**Comment by ijjk** [comment]
Hi, are you passing the `onClick` prop to the element you are rendering with your custom component as shown here in the docs https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-function-component? Using an `a` tag as the direct child of `next/link` helps ensure the most optimized experience since you don't have to worry about all the props being passed correctly

**Reply by rodolphonetto** [reply]
Thanks for your time Bro, I did chance the custom component for an `<a>` but It not solved :/

**Reply by ijjk** [reply]
Are the requests for `/_next` being served in your custom server properly? The hard reload is typically caused by either invalid `href` being provided to `next/link` (should be the page in your pages folder), the `onClick` prop not being passed correctly for a custom component, or the page's bundle failing to load from `/_next`

**Reply by rodolphonetto** [reply]
I've this there

```
  server.get("/_next/*", (req, res) => {
    handle(req, res)
  })
```

When I click a link this is the request

![Capturar](https://user-images.githubusercontent.com/18093520/88745075-809a2b00-d11f-11ea-88e3-d5f1e49cb8a4.JPG)

**Reply by ijjk** [reply]
If you can provide a repo with a reproduction I'd be happy to take a closer look
